api/modules/trips/views.py

```diff
@@ -237,3 +238,27 @@ def remove_user_from_trip(request, trip_id):
         return Response(str(e), status=status.HTTP_400_BAD_REQUEST)
 
     return Response(status=status.HTTP_200_OK)
+
+
+@api_view(['GET'])
+def get_trip_cities(request, user_id):
+    """
+    all the unique cities user has in his list of trips.
+    :param request:
+    :param user_id:
+    :return: 404 if user does not exist
+    :return: 200 successful
+    """
+    # if user with user_id does not exist
+    if not User.objects.filter(id=user_id).exists():
+        error_message = "User does not exists."
+        return Response(error_message, status=status.HTTP_404_NOT_FOUND)
+
+    city_id_list = []
+    trips = Trip.objects.filter(users=user_id)
+    for trip in trips:
+        city_id_list.append(trip.city.id)
```

Comment:
You can append `trip.city` instead of making a list of city IDs and then fetching each city by ID

Reply:
:+1: